cmd/influxd/inspect/verify_seriesfile.go

```diff
@@ -9,10 +9,90 @@ import (
 	"sort"
 	"sync"
 
+	"github.com/influxdata/influxdb/v2/logger"
 	"github.com/influxdata/influxdb/v2/tsdb"
+	"github.com/spf13/cobra"
 	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
 )
 
+type args struct {
+	dir        string
+	db         string
+	seriesFile string
+	verbose    bool
+	concurrent int
+}
+
+func NewVerifySeriesfileCommand() *cobra.Command {
+	var arguments args
+	cmd := &cobra.Command{
+		Use:   "verify-seriesfile",
+		Short: "Verifies the integrity of series files.",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SetOut(os.Stdout)
+
+			config := logger.NewConfig()
+			config.Level = zapcore.WarnLevel
+			if arguments.verbose {
+				config.Level = zapcore.InfoLevel
+			}
+			log, err := config.New(os.Stderr)
+			if err != nil {
+				return err
+			}
+
+			v := NewVerify()
+			v.Logger = log
+			v.Concurrent = arguments.concurrent
+
+			var db string
+			if arguments.seriesFile != "" {
+				db = arguments.seriesFile
+			} else if arguments.db != "" {
+				db = filepath.Join(arguments.dir, arguments.db, "_series")
+			}
+			if db != "" {
+				_, err := v.VerifySeriesFile(db)
+				return err
+			}
+
+			dbs, err := ioutil.ReadDir(arguments.dir)
+			if err != nil {
+				return err
+			}
+
+			for _, db := range dbs {
+				if !db.IsDir() {
+					continue
+				}
+				_, errNew := v.VerifySeriesFile(filepath.Join(arguments.dir, db.Name(), "_series"))
+				if errNew != nil {
+					v.Logger.Error("Failed to verify series file",
+						zap.String("filename", fmt.Sprintf(filepath.Join(arguments.dir, db.Name(), "_series"))),
+						zap.Error(errNew))
+					err = errNew
```

Comment:
This will cause the last error to be printed twice. Instead of that, a common pattern I've seen/used is to track a `bool` outside of the loop and:
1. Set the `bool` to `true` whenever a file fails verification (it's idempotent, so you can reset it to `true` over and over if needed)
2. At the end of the loop, return some generic error if the bool is `true`. i.e. `Some files failed verification, see logs for details`

Reply:
I implemented this, but also decided to keep the `err = errNew` line so that the full function will still return some error if any error was hit in this loop. If there's a better way to do this, possibly just returning some generic error, that might be better

Reply:
Yes, I'll comment a suggestion below